### PR TITLE
#99 InputType implementation changed with sealed class

### PR DESCRIPTION
**Pull request by kovalandrew**
Type changed, sample updated.
!!!! Required change current types usage from `InputType.PHONE` to `InputType.Phone()` !!!!

### CODE REVIEW

File: sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/sample/CryptoProfileSample.kt [hunk 1]
```diff
@@ -49,28 +49,28 @@ class CryptoProfileScreen(
...
-                        inputType = InputType.PHONE
+                        inputType = InputType.Phone()
                     )
                     +input(
                         size = WidgetSize.WidthAsParentHeightWrapContent,
                         id = Id.EmailInput,
                         label = const("Email"),
                         field = viewModel.emailField,
-                        inputType = InputType.EMAIL
+                        inputType = InputType.Phone()
```
Comment (Alex009) [inline sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/sample/CryptoProfileSample.kt]: email

File: widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/factory/SystemInputViewFactory.kt [hunk 2]
```diff
@@ -110,7 +110,7 @@ actual class SystemInputViewFactory actual constructor(
         if (widget.inputType?.mask != null) {
             val delegate = DefaultFormatterUITextFieldDelegate(
                 inputFormatter = DefaultTextFormatter(
-                    widget.inputType.mask.toIosPattern(),
+                    widget.inputType.mask?.toIosPattern() ?: "",
```
Comment (Alex009) [inline widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/factory/SystemInputViewFactory.kt]: if null then nothing formatter should be

File: widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/factory/FloatingLabelInputViewFactory.kt [hunk 3]
```diff
@@ -384,7 +384,7 @@ actual class FloatingLabelInputViewFactory actual constructor(
             if (inputType?.mask != null) {
                 formatterDelegate = DefaultFormatterUITextFieldDelegate(
                     DefaultTextFormatter(
-                        textPattern = inputType.mask.toIosPattern(),
+                        textPattern = inputType.mask?.toIosPattern() ?: "",
```
Comment (Alex009) [inline widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/factory/FloatingLabelInputViewFactory.kt]: if null then nothing formatter should be

File: sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/sample/CryptoProfileSample.kt (hunk 1 above)
Comment (kovalandrew) [reply sample/mpp-library/src/commonMain/kotlin/com/icerockdev/library/sample/CryptoProfileSample.kt]: Fixed

File: widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/factory/SystemInputViewFactory.kt (hunk 2 above)
Comment (kovalandrew) [reply widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/factory/SystemInputViewFactory.kt]: check added

File: widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/factory/FloatingLabelInputViewFactory.kt (hunk 3 above)
Comment (kovalandrew) [reply widgets/src/iosMain/kotlin/dev/icerock/moko/widgets/factory/FloatingLabelInputViewFactory.kt]: check added

**Pull request merged**